src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -1020,6 +1023,144 @@ void MeshForwarder::UpdateRoutes(uint8_t *           aFrame,
     return;
 }
 
+void MeshForwarder::UpdateFragmentLifetime(void)
+{
+    bool shouldRun = false;
+
+    for (size_t i = 0; i < OT_ARRAY_LENGTH(mFragmentEntries); i++)
+    {
+        if (mFragmentEntries[i].GetLifetime() != 0)
+        {
+            mFragmentEntries[i].DecrementLifetime();
+
+            if (mFragmentEntries[i].GetLifetime() != 0)
+            {
+                shouldRun = true;
+            }
+        }
+    }
+
+    if (shouldRun && !mUpdateTimer.IsRunning())
+    {
+        mUpdateTimer.Start(kStateUpdatePeriod);
+    }
+}
+
+void MeshForwarder::UpdateFragmentPriority(Lowpan::FragmentHeader &aFragmentHeader,
+                                           uint8_t                 aFragmentLength,
+                                           uint8_t                 aPriority)
+{
+    FragmentPriorityEntry *entry;
+
+    if (aFragmentHeader.GetDatagramOffset() == 0)
+    {
+        VerifyOrExit((entry = GetUnusedFragementPriorityEntry()) != NULL);
+
+        entry->SetDatagramTag(aFragmentHeader.GetDatagramTag());
+        entry->SetPriority(aPriority);
+        entry->SetLifetime(kReassemblyTimeout);
+
+        if (!mUpdateTimer.IsRunning())
+        {
+            mUpdateTimer.Start(kStateUpdatePeriod);
+        }
+    }
+    else
+    {
+        VerifyOrExit((entry = FindFragmentPriorityEntry(aFragmentHeader.GetDatagramTag())) != NULL);
+
+        if (aFragmentHeader.GetDatagramOffset() + aFragmentLength >= aFragmentHeader.GetDatagramSize())
```

Comment:
How about resetting the lifetime each time we receive a fragment that matches this entry? This way we can redefine `kReassemblyTimeout` to indicate the maximum time delay between fragments and use a smaller value.

Reply:
Changed.